### PR TITLE
fix bug mock reset

### PR DESCRIPTION
**Pull request by KevinLoiseau**
Hello,

This fix resolve all our problem explain in these issue : https://github.com/fog/fog-openstack/issues/59 and https://github.com/fog/fog-core/issues/178 .

I'm not sure it is the best response at the problem, but for us it works and i think it can help.

### CODE REVIEW

**Comment by seanhandley** [comment]
This is great @KevinLoiseau. If you fix up the styling issues highlighted by HoundCI then we'll review and merge

**Comment by plribeiro3000** [comment]
LGTM. Would you be able to move the requires to autoload as well please?

File: lib/fog/openstack.rb [hunk 1]
```diff
@@ -23,28 +24,37 @@
 require 'fog/openstack/planning'
 
 module Fog
+
+  module Compute
+    autoload :OpenStack, File.expand_path('../openstack/compute', __FILE__)
+  end
```
Comment (plribeiro3000) [inline lib/fog/openstack.rb]: Add line breaks between blocks to follow the standards please. Thanks.

**Comment by KevinLoiseau** [comment]
Thank you for your comments.
I have fix up the styling issues highlighted by HoundCI and delete useless require.

It's good for you ?

**Comment by plribeiro3000** [comment]
:+1:

**Comment by seanhandley** [comment]
Thanks @KevinLoiseau :+1:

**Pull request merged**